File: system-test/java/src/main/java/com/optivem/shop/dsl/driver/adapter/external/erp/client/ErpRealClient.java
package com.optivem.shop.dsl.driver.adapter.external.erp.client;

import com.optivem.shop.dsl.driver.adapter.external.erp.client.dtos.ExtCreateProductRequest;
import com.optivem.shop.dsl.driver.adapter.external.erp.client.dtos.error.ExtErpErrorResponse;
import com.optivem.shop.dsl.common.Result;

public class ErpRealClient extends BaseErpClient {
    private static final String PRODUCTS_ENDPOINT = "/api/products";

    public ErpRealClient(String baseUrl) {
        super(baseUrl);
    }

    public Result<Void, ExtErpErrorResponse> createProduct(ExtCreateProductRequest request) {
        return httpClient.post(PRODUCTS_ENDPOINT, request);
    }
}

File: system-test/typescript/src/testkit/driver/adapter/external/erp/client/ErpRealClient.ts
import { Result, success, failure } from '../../../../../common/result.js';
import type { ErpErrorResponse } from '../../../../port/external/erp/dtos/ErpErrorResponse.js';
import type { ReturnsProductRequest } from '../../../../port/external/erp/dtos/ReturnsProductRequest.js';
import { BaseErpClient } from './BaseErpClient.js';

export class ErpRealClient extends BaseErpClient {
  async createProduct(request: ReturnsProductRequest): Promise<Result<void, ErpErrorResponse>> {
    const response = await fetch(`${this.baseUrl}/api/products`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        id: request.sku,
        title: 'Test Product',
        description: 'Test Product Description',
        price: request.price,
        category: 'Test',
        brand: 'Test',
      }),
    });
    if (response.ok) return success(undefined);
    return failure({ message: `Failed to create product: ${response.status}` });
  }
}
